ortools/linear_solver/xpress_interface.cc

```diff
@@ -1261,11 +1263,16 @@ void XpressInterface::ExtractNewVariables() {
         CHECK_STATUS(XPRSaddcols(mLp, newcols, 0, obj.get(), cmatbeg.data(),
                                  cmatind.get(), cmatval.get(), lb.get(),
                                  ub.get()));
+        if (!colname.empty()) {
+          CHECK_STATUS(XPRSaddnames(mLp, XPRS_NAMES_COLUMN, colname.data(), 0,
+                                    newcols - 1));
+        }
```

Comment:
- `colname` will never be empty, since we add at least one '\0' character (assuming there is at least one column). Consider using existing flag `have_names`. If you do so, you'll need to fix a typo at l. 1162 (missing `!`)
- You should check l. 1246, there is another call to `XPRSaddcols`